js/maps.js

```diff
@@ -78,14 +78,31 @@ function buildLayerGroup(versionGroup) {
       htmlApproveButton.onclick = undefined;
       
       API.approve(eventName, versionGroup.version)
-        .then(function (res) {
-          if (!res.ok) {
-            throw 'General Error - server returned ' + res.status;
-          }
+        .then(function (response) {
+          // any reason to display the 201 response json here? Looks like:
+          // [{
+          // 	"name": "metadata",
+          // 	"url": "https://planetary-response-network.s3.amazonaws.com/events/dominica_2018_demo/layers/approved/v2/metadata.json"
+          // }, {
+          // 	"name": "overlay_0",
+          // 	"url": "https://planetary-response-network.s3.amazonaws.com/events/dominica_2018_demo/layers/approved/v2/overlay_0.csv"
+          // }, {
+          // 	"name": "overlay_1",
+          // 	"url": "https://planetary-response-network.s3.amazonaws.com/events/dominica_2018_demo/layers/approved/v2/overlay_1.csv"
+          // }, {
+          // 	"name": "overlay_2",
+          // 	"url": "https://planetary-response-network.s3.amazonaws.com/events/dominica_2018_demo/layers/approved/v2/overlay_2.csv"
+          // }, {
+          // 	"name": "overlay_3",
+          // 	"url": "https://planetary-response-network.s3.amazonaws.com/events/dominica_2018_demo/layers/approved/v2/overlay_3.csv"
+          // }]
+
+          // I think this has to reflect on the response obj here to decide what todo
+          // its either the body and all good
```

Comment:
I'm unsure how to switch the behaviour here, how does one reflect on the different return values?

Reply:
If you want to switch based on the response status, you'll need to do that in the API code: https://github.com/zooniverse/PRN-maps/blob/a64902e31db45028b36ca38e8abc5c25d926ef18/js/api.js#L14-L25

Reply:
I've updated the API post method to throw now so the calling code can catch.